pilot/pkg/serviceregistry/kube/controller.go

```diff
@@ -306,6 +306,28 @@ func (c *Controller) ManagementPorts(addr string) model.PortList {
 	return managementPorts
 }
 
+func getReadinessProbe(podSpec *v1.PodSpec) *model.Probe {
+	for _, container := range podSpec.Containers {
+		if container.ReadinessProbe != nil && container.ReadinessProbe.Handler.HTTPGet != nil {
+			return &model.Probe{
```

Comment:
This is getting probes from the 1st container which we happen to hit here.
There could be multiple container with health checks.

Also missing are, specifically Host and Port, otherwise we will insert these into all inbound listeners.